### DISCUSSION TITLE
Infinite space

### DISCUSSION THREAD

**Discussion by sunaina2821** [JoelForamitti/agentpy#43] [Q&A] [answered]
Hi,

I am using agentpy for a flocking model and need to implement infinite space. As mentioned in the source code of agentpy.space.py, there's a TO-DO for the exact thing that I want : # TODO Add option of space without shape (infinite). Any pointers on how I can approach this problem and perhaps even add it to the agentpy package? 

Thank you.

**Comment by jofmi** [comment, marked as answer]
Hi, sorry for the late answer! It should be quite easy to create an infinite space. Basically, shape has to become an optional argument, with a space without a shape being infinite. Then, all features of space that use the attribute shape (like the kdetree or border behavior) have to be disabled. If you would be interested to implement this, feel invited to make a pull request!
